### PR TITLE
Use TryGetObject instead of ContainsObject and GetObject

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Use TryGetObject instead of ContainsObject and GetObject.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8575)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 1]
```diff
@@ -1014,15 +1014,12 @@ public virtual ImageLayout BackgroundImageLayout
...
                 {
-                    return ImageLayout.Tile;
-                }
-                else
-                {
-                    return (ImageLayout)Properties.GetObject(s_backgroundImageLayoutProperty)!;
+                    return imageLayout;
                 }
+
+                return ImageLayout.Tile;
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:1022]:
```c#
get => Properties.TryGetObject(s_backgroundImageLayoutProperty, out ImageLayout imageLayout) ? imageLayout : ImageLayout.Tile;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCellStyle.cs [hunk 2]
```diff
@@ -148,12 +148,12 @@ public object DataSourceNullValue
...
             {
-                if (Properties.ContainsObject(PropDataSourceNullValue))
+                if (Properties.TryGetObject(PropDataSourceNullValue, out object value))
                 {
-                    return Properties.GetObject(PropDataSourceNullValue);
+                    return value;
                 }
 
-                return System.DBNull.Value;
+                return DBNull.Value;
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCellStyle.cs:156]: There are a few places in this change where we can simplify the code into the following form. Let's go ahead and do that to clean it up even more.
```c#
get => Properties.TryGetObject(PropDataSourceNullValue, out object value) ? value : DBNull.Value;
```

**Review by dmitrii-drobotov — APPROVED** [review]
Nice!

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs [hunk 3]
```diff
@@ -254,17 +254,17 @@ public virtual bool Frozen
...
         internal bool HasDefaultHeaderCellType
         {
-            get => Properties.ContainsObject(s_propDefaultHeaderCellType) && Properties.GetObject(s_propDefaultHeaderCellType) is not null;
+            get => Properties.TryGetObject(s_propDefaultHeaderCellType, out object? value) && value is not null;
         }
 
         internal bool HasHeaderCell
         {
-            get => Properties.ContainsObject(s_propHeaderCell) && Properties.GetObject(s_propHeaderCell) is not null;
+            get => Properties.TryGetObject(s_propHeaderCell, out object? value) && value is not null;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs:267]: Consider adding another helper that encapsulates the null-check in addition to TryGet. `IsNotNull` perhaps.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs [hunk 4]
```diff
@@ -254,17 +254,17 @@ public virtual bool Frozen
         [Browsable(false)]
         public bool HasDefaultCellStyle
         {
-            get => Properties.ContainsObject(s_propDefaultCellStyle) && Properties.GetObject(s_propDefaultCellStyle) is not null;
+            get => Properties.TryGetObject(s_propDefaultCellStyle, out object? value) && value is not null;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs:257]: Consider adding another helper that encapsulates the null-check in addition to TryGet. IsNotNull perhaps.

Sorry, I didn't save my comment and only noticed that it's in a pending state after the merge.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs:257]: I created https://github.com/dotnet/winforms/pull/8594 about this.

**Pull request merged**